dev/run-tests.py

```diff
@@ -0,0 +1,506 @@
+    in and returns the subprocess for any further processing"""
+
+    run_cmd(["./build/mvn"] + mvn_args)
+
+
+def exec_sbt(sbt_args=[]):
+    """Will call SBT in the current directory with the list of mvn_args passed
+    in and returns the subprocess for any further processing"""
+
+    sbt_cmd = ["./build/sbt"] + sbt_args
+
+    sbt_output_filter = re.compile("^.*[info].*Resolving" + "|" +
+                                   "^.*[warn].*Merging" + "|" +
+                                   "^.*[info].*Including")
+
+    # NOTE: echo "q" is needed because sbt on encountering a build file
+    # with failure (either resolution or compilation) prompts the user for
+    # input either q, r, etc to quit or retry. This echo is there to make it
+    # not block.
+    echo_proc = subprocess.Popen(["echo", "\"q\n\""], stdout=subprocess.PIPE)
+    sbt_proc = subprocess.Popen(sbt_cmd,
+                                stdin=echo_proc.stdout,
+                                stdout=subprocess.PIPE)
+    echo_proc.wait()
+    for line in iter(sbt_proc.stdout.readline, ''):
+        if not sbt_output_filter.match(line):
+            print line,
+    retcode = sbt_proc.wait()
+
+    if retcode > 0:
+        exit_from_command_with_retcode(sbt_cmd, retcode)
+
+
+def get_hadoop_profiles(hadoop_version):
+    """Return a list of profiles indicating which Hadoop version to use from
+    a Hadoop version tag."""
+
+    sbt_maven_hadoop_profiles = {
+        "hadoop1.0": ["-Phadoop-1", "-Dhadoop.version=1.0.4"],
+        "hadoop2.0": ["-Phadoop-1", "-Dhadoop.version=2.0.0-mr1-cdh4.1.1"],
+        "hadoop2.2": ["-Pyarn", "-Phadoop-2.2"],
+        "hadoop2.3": ["-Pyarn", "-Phadoop-2.3", "-Dhadoop.version=2.3.0"],
+    }
+
+    try:
+        hadoop_profiles = sbt_maven_hadoop_profiles[hadoop_version]
+    except KeyError:
+        print "[error] Could not find", hadoop_version, "in the list. Valid options",
+        print "are 'hadoop1.0', 'hadoop2.0', 'hadoop2.2', and 'hadoop2.3'."
+        sys.exit(int(os.environ.get("CURRENT_BLOCK", 255)))
+
+    return hadoop_profiles
+
+
+def get_build_profiles(hadoop_version="hadoop2.3",
+                       base_profiles=True,
+                       hive_profiles=False):
+    """Returns a list of hadoop profiles to be used as looked up from the passed in hadoop profile
+    key with the option of adding on the base and hive profiles."""
+
+    base_profiles = ["-Pkinesis-asl"]
+    hive_profiles = ["-Phive", "-Phive-thriftserver"]
+    hadoop_profiles = get_hadoop_profiles(hadoop_version)
+
+    build_profiles = hadoop_profiles
+    # first, check and add the base profiles
+    if base_profiles:
+        build_profiles = build_profiles + base_profiles
+    # second, check and add the hive profiles
+    if hive_profiles:
+        build_profiles = build_profiles + hive_profiles
+
+    return build_profiles
+
+
+def build_spark_maven(hadoop_version):
+    build_profiles = get_build_profiles(hadoop_version, hive_profiles=True)
```

Comment:
Maybe add a comment on the previous line to explain that we always build with Hive support, even if we end up skipping the Hive tests in most PRB builds?